server/ComputeSuperpixelFeatures/ComputeSuperpixelFeatures.py

```diff
@@ -0,0 +1,387 @@
+from skimage.measure import regionprops
+from skimage.segmentation import slic
+from keras.models import Model, load_model
+
+import os
+import sys
+import json
+import time
+
+import numpy as np
+import dask
+import h5py
+
+import histomicstk.preprocessing.color_normalization as htk_cnorm
+import histomicstk.preprocessing.color_deconvolution as htk_cdeconv
+import histomicstk.utils as htk_utils
+
+import large_image
+
+from ctk_cli import CLIArgumentParser
+
+import logging
+logging.basicConfig(level=logging.CRITICAL)
+
+sys.path.append(os.path.normpath(os.path.join(os.path.dirname(__file__), '..')))
+from cli_common import utils as cli_utils  # noqa
+
+
+def compute_input_tensors(slide_path, tile_position, args, **it_kwargs):
+
+    # get slide tile source
+    ts = large_image.getTileSource(slide_path)
+
+    # get current magnification
+    magnification = ts.getMetadata()['magnification']
+
+    # get requested tile
+    tile_info = ts.getSingleTile(
+        tile_position=tile_position,
+        format=large_image.tilesource.TILE_FORMAT_NUMPY,
+        **it_kwargs)
+
+    # get tile image
+    im_tile = tile_info['tile'][:, :, :3]
+
+    # get current left and top positions
+    tile_left_position = tile_info['gx']
+    tile_top_position = tile_info['gy']
+
+    # perform color normalization
+    im_nmzd = htk_cnorm.reinhard(im_tile,
+                                 args.reference_mu_lab,
+                                 args.reference_std_lab)
+
+    # perform color decovolution
+    w = cli_utils.get_stain_matrix(args)
+
+    im_stains = htk_cdeconv.color_deconvolution(im_nmzd, w).Stains
+
+    im_hematoxylin_stain = im_stains[:, :, 0].astype(np.float)/255.0
+    im_eosin_stain = im_stains[:, :, 1].astype(np.float)/255.0
+
+    # get the number of super-pixels
+    im_width, im_height = im_tile.shape[:2]
+    n_superpixels = (im_width/args.patchSize) * (im_height/args.patchSize)
+
+    # perform SLIC to get labels with max compactness
+    # slic has a value 0 as a first label, so added 1
+    im_label = slic(im_nmzd, n_segments=n_superpixels, compactness=100) + 1
```

Comment:
Expose compactness as a parameter to the CLI with a reasonable default value that works on multiple slides.

Reply:
This parameter is a bit abstract - at the least we need to describe what the effect of increasing or decreasing is on the superpixels